libs/ledger-live-common/src/hw/actions/app.ts

```diff
@@ -489,6 +503,12 @@ export const createAction = (
               firmwareResolvedRef.current = true;
             } else if (e.type === "has-outdated-app") {
               outdatedAppRef.current = e.outdatedApp as AppAndVersion;
+            } else if (
+              e.type === "deprecation" &&
+              "onContinue" in e &&
+              typeof e.onContinue === "function"
```

Comment:
```suggestion
              e.type === "deprecation"
```
This is enough for the type checking